src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -589,7 +589,7 @@ private static void ApplyLibContract(LocalPackageInfo package, LockFileTargetLib
                              && d.SuppressParent != LibraryIncludeFlags.All
                              && d.ReferenceType == LibraryDependencyReferenceType.Direct)
                     .Select(d => GetDependencyVersionRange(d))
-                    .ToList()
+                    .AsList()
```

Comment:
This change is unnecessary. There's no functional change here.
LINQ generates enumerates which had to be converted to Lis